### ISSUE TITLE
XLA_SAVE_TENSORS_FILE flag slows performance significantly

### ISSUE THREAD

**Issue by mikowals** [tensorflow/swift-apis#952]
Adding `setenv("XLA_SAVE_TENSORS_FILE", "file", 1)` to Colab TPU instances appears to triple the execution time of training.

For example, altering the first cell of the [X10 tutorial](https://colab.research.google.com/github/tensorflow/swift/blob/master/docs/site/tutorials/introducing_x10.ipynb#scrollTo=kZRlD4utdPuX) to:

    %install '.package(url: "https://github.com/tensorflow/swift-models", .branch("tensorflow-0.9"))' Datasets ImageClassificationModels
    setenv("XLA_SAVE_TENSORS_FILE", "file", 1)
    print("\u{001B}[2J")

And running running the X10 LeNet example takes performance from 7.7 seconds in the first epoch to 25.9 seconds.  In later epochs performance time remains nearly tripled going from 6.8 seconds to 18.2.  

This tripling is consistent with a larger ResNet which went from 60 seconds to 180 seconds per epoch on TPU and from 90 seconds to 250 seconds on GPU.

- Is that kind of slowdown to be expected  and something that should be mentioned in docs?
- Does using it along with `PrintX10Metrics()` impact the "Accumulator" line of each metric reported so it doesn't make sense to use `PrintX10Metrics()` and XLA_SAVE_TENSORS_FILE flag together?

**Comment by asuhan** [comment]
Yes, that type of slowdown is to be expected. There is a fair amount of additional work which needs to happen when `XLA_SAVE_TENSORS_FILE` is set: serializing graphs to a human-readable representation and writing them to file. I'll document it.

Neither `PrintX10Metrics` nor exporting `XLA_SAVE_TENSORS_FILE` impact counters, so there's no harm in using them together.
